src/engine/target.js

```diff
@@ -118,6 +119,44 @@ class Target extends EventEmitter {
     }
 
     /**
+     * Creates a variable with information given in the event and adds it to
+     * the dictionary of variables.
+     * @param {object} e Blockly "variable" event.
+     */
+    createVariable (e) {
```

Comment:
Could you change the arguments to these so that they do not contain the blockly event, but instead the stuff that will be used from the event. E.g.

```
createVariable(id, name) { ...
renameVariable(id, newName) { ...
deleteVariable(id) { ...
```